drivers/serial/uart_nrfx_uarte.c

```diff
@@ -643,17 +711,20 @@ static int uarte_instance_init(struct device *dev,
 	#endif /* CONFIG_UART_1_INTERRUPT_DRIVEN */
 
 	#ifdef CONFIG_UART_1_NRF_FLOW_CONTROL
-		#define UARTE_1_NRF_HWFC_CONFIG	       UARTE_NRF_HWFC_ENABLED(1)
-		#ifndef DT_NORDIC_NRF_UARTE_UART_1_RTS_PIN
-		#error Flow control for UARTE1 is enabled, but RTS pin is not defined.
-		#endif
-		#ifndef DT_NORDIC_NRF_UARTE_UART_1_CTS_PIN
-		#error Flow control for UARTE1 is enabled, but CTS pin is not defined.
-		#endif
+		#define UARTE_1_NRF_HWFC_CONFIG	NRF_UARTE_HWFC_ENABLED
 	#else
-		#define UARTE_1_NRF_HWFC_CONFIG	       UARTE_NRF_HWFC_DISABLED
+		#define UARTE_1_NRF_HWFC_CONFIG	NRF_UARTE_HWFC_DISABLED
 	#endif /* CONFIG_UART_1_NRF_FLOW_CONTROL */
 
+	#if defined(DT_NORDIC_NRF_UARTE_UART_1_RTS_PIN) && \
+	    defined(DT_NORDIC_NRF_UARTE_UART_1_CTS_PIN)
+		#define UARTE_1_NRF_RTS_CTS_PINS UARTE_NRF_RTS_CTS_SET(1)
+		#define UARTE_1_CONFIG_RTS_CTS .rts_cts_pins_set = true
+	#else
+		#define UARTE_1_NRF_RTS_CTS_PINS UARTE_NRF_RTS_CTS_NOT_SET
+		#define UARTE_1_CONFIG_RTS_CTS .rts_cts_pins_set = false
+	#endif
+
 	#ifdef CONFIG_UART_1_NRF_PARITY_BIT
 		#define UARTE_1_NRF_PARITY_BIT	       NRF_UARTE_PARITY_INCLUDED
 
```

Comment:
Could you remove this extra empty line? Since you are applying some other corrections here, it seems to be a good occasion for doing this.